### ISSUE TITLE
What's the difference among different DEYO version?

### ISSUE THREAD

**Issue by avenger123456** [ouyanghaodong/DEYO#3]
Great work! Hi, I observe your previous DEYO, DEYOv2, DEYOv3 papers，and this DEYO. What's the difference among these versions of DEYO? I am interested in how you iterate to this final version.

**Comment by ouyanghaodong** [comment]
In the initial experiment, we assessed the model's ideal performance by matching YOLOv5's output with the ground truth on a one-to-one basis. Despite YOLOv5's AP being lower than DINO-4scale's, we found that, in an ideal scenario, YOLOv5's performance significantly outperformed DINO-4scale. We believe this difference stems from the performance bottlenecks of each: YOLOv5 is primarily limited by classification accuracy, while DINO-4scale is constrained by positioning accuracy. Based on this discovery, we developed the initial version of DEYO. However, DEYO relies on NMS due to the transformer's inability to directly handle dense queries. By adopting greedy matching and rank feature, we effectively addressed this issue, leading to the development of DEYOv2. Despite this, DEYOv2's design was not elegant, and we lacked sufficient GPU resources to train a larger-scale DEYOv2. This motivated the development of a lightweight DEYOv3. We discovered that conventional DETRs training methods were not suitable for DEYOv3, and it was impossible to apply DEYOv2's strategies to DEYOv3. Therefore, we developed a step-by-step training strategy, which significantly enhanced DEYO's performance. Regrettably, perhaps because DEYOv3's performance bottleneck lies in classification accuracy rather than positioning accuracy, DEYOv2's strategies were ineffective. We then focused on improving the step-by-step training to minimize training overhead, ultimately resulting in this version of DEYO.
